--- a/lib/DebugInfo/DWARFContext.cpp	2014-06-13 04:24:39.000000000 +0200
+++ b/lib/DebugInfo/DWARFContext.cpp	2014-12-18 14:06:47.000000000 +0100
@@ -738,8 +738,10 @@
           if (ec) {
             errs() << "Aaaaaa! Nameless relocation! Aaaaaa!\n";
           }
-          errs() << "error: failed to compute relocation: "
-                 << Name << "\n";
+          // RoboVM note: This happens for RoboVM compiled code and also with code compiled with clang.
+          //              Prevent this from being written to stderr.
+          //errs() << "error: failed to compute relocation: "
+          //       << Name << "\n";
           continue;
         }
 
